# CSE110 Lab Assignment 3

This Assignment is to help you develop your concept of Strings in Python.







##Write your name, student ID and CSE110 theory section number below:







In [ ]:
#STUDENT NAME: Aritra Ghose
#STUDENT ID: x
#CSE110 THEORY SECTION: 03

## Write the code in Python to do the following tasks:

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**


### Task 1

Write a Python program that will ask the user to input a string (containing exactly one word). Then your program should print subsequent substrings of the given string as shown in the examples below.

=====================================================================

**Example 1:**\
Input: BANGLA\
Output:\
B\
BA\
BAN\
BANG\
BANGL\
BANGLA

=====================================================================

**Example 2:**\
Input: DREAM\
Output:\
D\
DR\
DRE\
DREA\
DREAM

=====================================================================

**Hints(1):** Need to use "for loop" for this task.

**Hints(2):** Need to use print() function for printing newlines.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

We need use print(end = "") to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the following output right next to the previous one)\
12

=====================================================================






In [ ]:
#to do

word = input("Enter a word: ")

for i in range(len(word)):
  print(word[0:i+1])


Enter a word: WORD
W
WO
WOR
WORD


In [ ]:
# 2

word = input("Enter a word: ")

highest_index = len(word)-1
index = 0

while (index <= highest_index):
  horizontal_index = 0
  while (horizontal_index <= index):
    print(word[horizontal_index], end = "")
    horizontal_index += 1
  index = index + 1
  print("")



Enter a word: SPAIN
S
SP
SPA
SPAI
SPAIN


### Task 2

Write a Python program that will ask the user to enter a  word as an input.

* If the length of the input string is less than 4, then your program should print the same string as an output.
* If the input string’s length is greater than 3, then your program should add "er" at the end of the input string.
* If the input string already ends with "er", then add "est" instead.
* If the input string already ends with "est", then your program should print the same input string as an output.

=====================================================================

**Example 1:**\
Input: strong\
Output: stronger

=====================================================================

**Example 2:**\
Input: stronger\
Output: strongest

=====================================================================

**Example 3:**\
Input: strongest\
Output: strongest

=====================================================================

**Example 4:**\
Input: abc\
Output: abc

=====================================================================

In [ ]:
#to do

word = input("Enter a word: ")

if (len(word) < 4):
  print(word)

elif (len(word) > 3):
  if word.endswith("est"):
    print(word)
  elif word.endswith("er"):
    print(word.replace("er", "est"))
  elif (not word.endswith("er")):
    print(word+"er")



Enter a word: strong
stronger


### Task 3

Write a Python program that will ask the user to input a string (containing exactly one word). Then print the ASCII code for each character in the String using the ord() function.

To check if your program is working correctly or not, you can find a list of all correct values from the following website. Look at “Dec” and “Char” columns only, ignore other columns.\
link: http://www.asciitable.com/

=====================================================================

**Example 1:**\
Input: Programming\
Output:\
P : 80\
r : 114\
o : 111\
g : 103\
r : 114\
a : 97\
m : 109\
m : 109\
i : 105\
n : 110\
g : 103

=====================================================================

**Example 2:**\
Input: hunger\
Output:\
h : 104\
u : 117\
n : 110\
g : 103\
e : 101\
r : 114

=====================================================================

In [ ]:
#to do

word = input("Enter a string: ")

for i in range(len(word)):
  print(word[i], ":", ord(word[i]))



Enter a string: hunger
h : 104
u : 117
n : 110
g : 103
e : 101
r : 114


### Task 4

Take a string as a input from the user with all small letters. Then print the next alphabet in sequence for each alphabet found in the input.

=====================================================================


*Hint: You need to use functions ord() and chr(). The ASCII value of ‘a’ is 97 and ‘z’ is 122.*

=====================================================================

**Example 1:**\
Input: abcd\
Output: bcde

=====================================================================

**Example 2:**\
Input: the cow\
Output: uif!dpx

=====================================================================

**Example 3: <font color='red'>[Must fulfil this criteria]</font>**\
Input: xyzabc\
Output: yzabcd

=====================================================================


In [ ]:
#to do

the_string = input("Enter a string containing small letters: ")

for i in the_string:
  # we are only considering the small letters
  if (ord(i) == 122):
    ascii_value = 97
    final_string = chr(ascii_value)
    print(final_string, end = "")

  else:
    ascii_value = ord(i) + 1
    final_string = chr(ascii_value)
    print(final_string, end = "")



Enter a string containing small letters: xyzabc
yzabcd

### Task 5

Write a Python program that will ask the user to enter two strings (s1, s2) as an input. Then create a mixed string with alternative characters from each string. Any leftover chars will be appended at the end of the resulting string.

=====================================================================

*Hint: For adding the leftover characters you may use string slicing.*

=====================================================================

**Example 1:**\
Input:\
"ABCD"\
"efgh"

Output:\
AeBfCgDh

=====================================================================

**Example 2:**\
Input:\
"ABCDENDFGH"\
"ijkl"

Output:\
AiBjCkDlENDFGH

=====================================================================

**Example 3:**\
Input:\
"ijkl"\
"ABCDENDFGH"

Output:\
iAjBkClDENDFGH

=====================================================================


In [ ]:
#todo

s1 = input("Enter the first string: ")
s2 = input("Enter the second string: ")
index_for_s2 = 0
total_index_s2 = len(s2) - 1

for i in s1:
  print(i, end="")
  print(s2[index_for_s2], end = "")
  index_for_s2 += 1

print(s2[index_for_s2:], end="")


Enter the first string: 12345
Enter the second string: ABCDEFGHIJ
1A2B3C4D5EFGHIJ

In [ ]:
# 2

s1 = input("Enter the first string: ")
s2 = input("Enter the second string: ")
index_for_s2 = 0
total_index_s2 = len(s2) - 1

for i in s1:
  print(i, end="")
  print(s2[index_for_s2], end = "")
  index_for_s2 += 1

while (index_for_s2 <= total_index_s2):
  print(s2[index_for_s2], end="")
  index_for_s2 += 1



Enter the first string: 12345
Enter the second string: abcdefghi
1a2b3c4d5efghi

### Task 6
Given a string, create a new string with all the consecutive duplicates **removed**.

=====================================================================

**Hint:** You may make a new string to store the result. You can check whether the current character and the next character are the same, then add that character to the new string.

=====================================================================


**Sample Input**:\
AAABBBBCDDBBECE

**Sample Output:**\
ABCDBECE

=====================================================================


In [ ]:
#to do

the_string = input("Enter a string: ")

for i in range(0, len(the_string)):
  if i == 0:
    print(the_string[0], end="")
  elif i!=0 and the_string[i]==the_string[i-1]:
    pass
  else:
    print(the_string[i], end="")


Enter a string: ABCDE
ABCDE

### Task 7

Write a python program that takes 2 inputs from the user, where the first input is a string with length greater than 1. The second input is the index of the first given string from where you have to start reversing. Then print the new string back to the user.

=====================================================================

**Sample Input 1:**

72418

4

**Sample Output 1:**

81427


**Explanation**: Our second input, index '4' is the last index of our first input String '72418', hence the entire string is reversed giving us '81427'.

=====================================================================

**Sample Input 2:**

12345

2

**Sample Output 2:**

32145

**Explanation**: The second input is '2' so we have to reverse from index 2 of our first input. The 2nd index of our first input String is '3', index 1 is '2' and index 0 is '1'. Hence, if we reverse indexes 0 to 2, we get '321'. Index 3 and 4 which is '4' and '5' respectively remains unchanged hence our final output is '32145'.

=====================================================================

**Sample Input 3:**

aBcd1234defg

5

**Sample Output 3:**

21dcBa34defg

**Explanation**: From our first input String 'aBcd1234defg',

index 0 = 'a'

index 1 = 'B'

index 2 = 'c'

index 3 = 'd'

index 4 = '1'

index 5 = '2'

index 6 = '3'

Since our second input is 5, index 0 to index 5 is reversed and we have '21dcBa' and the rest is unchanged from indexes 6 to 11 ('34defg'). Therefore, we have '21dcBa34defg' finally.

=====================================================================


In [ ]:
#to do

the_string = input("Enter a string with length greater than 1: ")
reverse_point = input("Enter the second string: ")

reverse_point = int(reverse_point)

for i in range(reverse_point, -1, -1):
  print(the_string[i], end="")

if (reverse_point < len(the_string)):
  print(the_string[reverse_point+1:])



Enter a string with length greater than 1: aBcd1234defg
Enter the second string: 5
21dcBa34defg


### Task 8

Write a python program that splits a given string on a given split character. The first input is a String and the second input is the character that will be used to split the first String. <font color='red'>[You cannot use the built-in split function]</font>

=====================================================================

**Sample Input 1:**<br/>
This-is-CSE110<br/>
-<br/>

**Sample Output 1:**<br/>
This<br/>
is<br/>
CSE110

**Explanation**: The second input which is the character '-', is used to split or divide the first input String 'This-is-CSE110' into 'This', 'is' and 'CSE110' which are printed individually in seperate lines.

=====================================================================

**Sample Input 2**:
tom@gmail,harry@yahoo,bob@gmail,mary@gmail

,

**Sample Output 2**:

tom@gmail

harry@yahoo

bob@gmail

mary@gmail

=====================================================================

In [ ]:
#to do

the_string = input("Enter the string: ")
the_character = input("Enter the character: ")

for i in range(len(the_string)):
  if (the_string[i]==the_character):
    print("")
  elif (the_string[i]!=the_character):
    print(the_string[i], end="")



Enter the string: This is CSE110
Enter the character:  
This
is
CSE110

### Task 9

Suppose you are given two strings, s1, and s2. Now, print a new string made up of the last characters and then the first characters of the input strings.

=====================================================================

**Sample Input 1**:

s1 = new

s2= string

**Sample Output 1**:

gwsn

**Explanation:** The last character of the String s2 is 'g'. The last character of the String s1 is 'w'. The first character of the String s2 is 's'. The first character of the String s1 is 'n'. Together they give us the ouput we want 'gwsn'.

=====================================================================

**Sample Input 2**:

s1 = abcd

s2= efgh

**Sample Output 2**:

hdea

**Explanation:** The last characters of the Strings s2 and s1 is 'h' and 'd' respectively while the first characters of the Strings is 'e' and 'a' respectively. Together they give us the ouput we want 'gwsn'.

=====================================================================


In [ ]:
#to do

s1 = input("Enter the first string: ")
s2 = input("Enter the second string: ")

print(s2[-1], end="")
print(s1[-1], end="")
print(s2[0], end="")
print(s1[0], end="")


Enter the first string: new
Enter the second string: string
gwsn

### Task 10

Write a Python program that takes a String as input from the user, removes the characters at even index and prints the resulting String in uppercase.

=====================================================================

**Sample Input 1**:

String

**Sample Output 1**:

TIG

**Explanation**: The characters 'S', 'r' and 'n' are at index positions 0, 2, and 4 respectively. Hence they are removed and the remaining characters 'tig' are capitalized giving us output 'TIG'.

=====================================================================

**Sample Input 2**:

abcd

**Sample Output 2**:

BD

=====================================================================


In [ ]:
#to do
the_string = input("Enter the string: ")
new_string = ""

for i in range(1, len(the_string), 2):
  new_string += the_string[i]

print(new_string.upper())

Enter the string: String
TIG


### Task 11

Write a python program that takes two inputs. The first input is a string and the second input is a number. If the number is even then concatenate the given string two times the given number and if the number is odd then concatenate the given string three times the given number.

=====================================================================

**Sample Input 1**:

CSE110

4

**Sample output 1**:

CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110

**Explanation**: The second input which is the number 4 is even, therefore the first string input 'CSE110' is concatenated(joined together) 4*2 = 8 times.

=====================================================================

**Sample Input 2**:

CSE110

3

**Sample Output 2**:

CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110

=====================================================================


In [ ]:
#to do

the_string = input("Enter the string: ")
the_number = int(input("Enter the number: "))

if (the_number%2 == 0):
  print(the_string*the_number*2)
else:
  print(the_string*the_number*3)


Enter the string: CSE110
Enter the number: 4
CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110


### Task 12

Write a python program that takes 2 inputs from the user. The first input is a string and the second input is a letter. The program should remove all occurences of the letter from the given string and print the output. If the letter is not found in the string and the length of string is greater than 3, then remove the first letter and last letter of the given string and print it. Otherwise print the string as it is. [You can assume that all the input will be in lowercase letter]

=====================================================================


**Sample Input 1**:

tanjiro kamado

a

**Sample output 1**:

tnjiro kmdo

**Explanation**: All 3 instances of the character 'a' is removed from the input String 'tanjiro kamado' to give us output 'tnjiro kmdo'.

=====================================================================

**Sample Input 2**:

eren yeager

k


**Sample Output 2**:

ren yeage

**Explanation**: The character 'k' is absent in the first input String 'eren yeager' and it's length is 11 which is greater than 3 therefore the first character 'e' and the last character 'r' is removed. Hence, the final String is 'ren yeage'.

=====================================================================

Sample Input 3:

hi

a

Sample Output 3:

hi

**Explanation**: The letter 'a' is not found in our first input 'hi', the length of which is 2. Since the character is not present and the length is less than 3, we print the String 'hi' as it is.

=====================================================================


In [ ]:
#to do

the_string = input("Enter the string: ")
the_letter = input("Enter the letter: ")

if the_letter in the_string:
  for i in range(len(the_string)):
    if the_string[i] != the_letter:
      print(the_string[i], end="")
    else:
      pass

else:
  if len(the_string) > 3:
    print(the_string[1:-1])
  else:
    print(the_string)


Enter the string: hi
Enter the letter: a
hi


## Optional Tasks (13 - 15) [Ungraded]

### Task 13


Write a python program that takes a string as an input from the user and then modifies the string in such a way that the string always starts with an uppercase letter and the case of each subsequent letter is the opposite of the previous letter (uppercase character followed by a lowercase character followed by an uppercase character and so on).  Finally the modified string is printed to show the user.

*Hints:  Flags/counters can be used to manage uppercase-lowercase.*

=====================================================================

**Sample Input 1:**\
Python programming is very easy

**Sample Output 1:**\
PyThOn PrOgRaMmInG iS vErY eAsY

=====================================================================

**Sample Input 2:**\
I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     love       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Python &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Programming

**Sample Output 2:**\
I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     lOvE       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pYtHoN&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pRoGrAmMiNg

=====================================================================

**Sample Input 3:**\
CSE110 Course

**Sample Output 3**\
CsE110 cOuRsE



=====================================================================

**Sample Input 4:**\
c

**Sample Output 4:**\
C

=====================================================================


In [ ]:
#to do

#Unsolved

the_string = input("Enter the string: ")

for i in range(0, len(the_string)):
  if i%2 == 0:
    print(the_string[i].upper(), end = "")
  else:
    print(the_string[i].lower(), end = "")


Enter the string: CSE110 Course
CsE110 cOuRsE

### Task 14
An anagram is a play on words created by rearranging the letters of the original word to make a new word or phrase.
So we can say two words are anagrams if they contain all of the same letters, but in a different order.


Write a python program that takes two strings from the user and tells if they are anagram or not.

=====================================================================

<b>Sample Input 1: </b><br/>
dusty<br/>
study<br/>
<b>Sample Output 1: </b> <br/>
They are anagram<br/>

=====================================================================

<b>Sample Input 2: </b><br/>
dustyyy<br/>
study<br/>
<b>Sample Output 2: </b><br/>
They are not anagram

In [ ]:
# to do

string_1 = input("Enter the first word: ")
string_2 = input("Enter the second word: ")
prev_string_2 = string_2
index = 0
highest_index = len(string_1)-1


while (index <= highest_index):
  if string_1[index] in string_2:
    the_substring = string_1[index]
    string_2 = string_2.replace(the_substring, "")
  else:
    pass
  index += 1

if len(string_2) == 0 and len(string_1)==len(prev_string_2):
  print("They are anargram")
else:
  print("They are not anargram")


Enter the first word: abcde
Enter the second word: deabcc
They are not anargram


###Task 15

Write a python program thats prints the largest consecutive sequence from a given string S. For example, if S = ‘AABBBCCAAAA’, print ‘AAAA’. If more than one largest consecutive sequence exists, print the first one. Print NONE if no consecutive sequence is found.

=====================================================================

**Sample Input 1:**<br/>
AABBBCCAAAA<br/>
**Sample Output 1:**<br/>
AAAA<br/>

=====================================================================

**Sample Input 2:**<br/>
AABBCC<br/>
**Sample Output 2:**<br/>
AA<br/>

=====================================================================

**Sample Input 3:**<br/>
ABCD<br/>
**Sample Output 3:**<br/>
NONE<br/>

=====================================================================

In [ ]:
#to do

# Unsolved

S = input("Enter the string: ")
